src/rnnlm/rnnlm-utils-test.cc

```diff
@@ -0,0 +1,187 @@
+// rnnlm/rnnlm-utils-test.cc
+
+#include <math.h>
+#include "rnnlm/rnnlm-utils.h"
+
+namespace kaldi {
+namespace rnnlm {
+
+void UnitTestSampleConvergence() {
+  // randomly generate unigrams
+  int n = rand() % 10000 + 1000;
+  vector<BaseFloat> selection_probs;
+  BaseFloat prob = 0;
+  BaseFloat prob_sum = 0;
+  for (int i = 0; i < n; i++) {
+    prob = RandUniform();
+    prob_sum += prob;
+    selection_probs.push_back(prob);
+  }
+  // KALDI_LOG << "Before Normalize, probs are:";
+  for (int i = 0; i < n; i++) {
+    selection_probs[i] /= prob_sum;
+    // KALDI_LOG << selection_probs[i];
+  }
+  // generate a outputs_set with ones and with size < k
+  int ones_size = rand() % (n / 2) + 1;
+  // KALDI_LOG << "Size of ones_set is " << ones_size;
+  std::set<int> outputs_set;
+  for (int i = 0; i < ones_size; i++) {
+    outputs_set.insert(rand() % n);
+  }
+  // generate a random number k from 1 to n
+  int k = 2 * ones_size;
```

Comment:
but I have to make sure ones_size < k right?
For selecting n from n, your code line 241 of rnnlm-utils.cc fails: 
ASSERTION_FAILED (NormalizeVec():rnnlm-utils.cc:241) : 'b < 1'
Other special cases like k = 1 or 2 can pass the test.